edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/metrics/AvaliabilityMetrics.cs

```diff
@@ -7,22 +7,28 @@ namespace Microsoft.Azure.Devices.Edge.Agent.Core.Metrics
     using System.IO;
     using System.Linq;
     using System.Runtime.CompilerServices;
+    using System.Threading.Tasks;
     using Microsoft.Azure.Devices.Edge.Util;
     using Microsoft.Azure.Devices.Edge.Util.Metrics;
     using Microsoft.Extensions.Logging;
 
     public interface IAvailabilityMetric
     {
         void ComputeAvailability(ModuleSet desired, ModuleSet current);
+        void OnCleanShutdown();
     }
 
-    public class AvailabilityMetrics : IAvailabilityMetric
+    public class AvailabilityMetrics : IAvailabilityMetric, IDisposable
     {
         readonly IMetricsGauge running;
         readonly IMetricsGauge expectedRunning;
         readonly ISystemTime time;
         readonly ILogger log = Logger.Factory.CreateLogger<Availability>();
 
+        // This allows edgeAgent to track its own avaliability. If edgeAgent shutsdown unexpectedly, it can look at the last checkpoint time to determine its previous avaliability.
+        readonly TimeSpan checkpointFrequency = TimeSpan.FromMinutes(5);
```

Comment:
should this be configurable?

Reply:
considering the reconcile frequency isn't configurable, I don't think this needs to be